rules/Line.js

```diff
@@ -0,0 +1,154 @@
+exports.missThreshold = parseFloat(hoot.get("generic.line.miss.threshold"));
+exports.reviewThreshold = parseFloat(hoot.get("generic.line.review.threshold"));
+exports.searchRadius = parseFloat(hoot.get("search.radius.generic.line"));
+exports.tagThreshold = parseFloat(hoot.get("generic.line.tag.threshold"));
+exports.baseFeatureType = "Line";
+exports.geometryType = "line";
+
+var angleHistogramExtractor = new hoot.AngleHistogramExtractor();
+var weightedShapeDistanceExtractor = new hoot.WeightedShapeDistanceExtractor();
+var distanceScoreExtractor = new hoot.DistanceScoreExtractor();
+var lengthScoreExtractor = new hoot.LengthScoreExtractor();
+var sublineMatcher = new hoot.MaximalSublineStringMatcher({
+    "way.matcher.max.angle": hoot.get("generic.line.matcher.max.angle"),
+    "way.subline.matcher": hoot.get("generic.line.subline.matcher")});
+
+/**
+ * Returns true if e is a candidate for a match. Implementing this method is
+ * optional, but may dramatically increase speed if you can cull some features
+ * early on. E.g. no need to check nodes for a polygon to polygon match.
+ *
+ * exports.matchCandidateCriterion takes precedence over this function and must
+ * be commented out before using it.
+ */
+exports.isMatchCandidate = function(map, e)
+{
+  return isLinear(e) && !isSpecificallyConflatable(map, e);
+};
+
+/**
+ * If this function returns true then all overlapping matches will be treated
+ * as a group. For now that means if two matches overlap then the whole group
+ * will be marked as needing review.
+ *
+ * If this function returns false the conflation routines will attempt to 
+ * pick the best subset of matches that do not conflict.
+ */
+exports.isWholeGroup = function() 
+{
+  return false;
+};
+
+/**
+ * Returns the match score for the three class relationships.
+ * - match
+ * - miss
+ * - review
+ *
+ * The scores should always sum to one. If they don't you will be taunted 
+ * mercilessly and we'll normalize it anyway. :P
+ */
+exports.matchScore = function(map, e1, e2) 
+{
+  var result = { miss: 1.0 };
+
+  if (e1.getStatusString() == e2.getStatusString()) 
+  {
+    return result;
+  }
+
+  hoot.trace("e1: " + e1.getId() + ", " + e1.getTags().get("name"));
+  if (e1.getTags().get("note"))
+  {
+    hoot.trace("e1 note: " + e1.getTags().get("note"));
+  }
+  hoot.trace("e2: " + e2.getId() + ", " + e2.getTags().get("name"));
+  if (e2.getTags().get("note"))
+  {
+    hoot.trace("e2 note: " + e2.getTags().get("note"));
+  }
+
+  var typeScore = getTypeScore(map, e1, e2);
+  var typeScorePassesThreshold = false;
+  if (typeScore >= exports.tagThreshold)
+  {
+    typeScorePassesThreshold = true;
+  }
+  hoot.trace("typeScore: " + typeScore);
+  hoot.trace("typeScorePassesThreshold: " + typeScorePassesThreshold);
+
+  // extract the sublines needed for matching - Note that this was taken directly from Highway.js.
+  var sublines = sublineMatcher.extractMatchingSublines(map, e1, e2);
```

Comment:
The geometry matching ranges from Weka derived rules brought over from existing scripts to simple geometry to geometry distance checks (Point.js, etc.).